pkg/client/restclient/request.go

```diff
@@ -811,7 +811,18 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 			r.backoffMgr.UpdateBackoff(r.URL(), err, resp.StatusCode)
 		}
 		if err != nil {
-			return err
+			if !net.IsConnectionReset(err) {
+				return err
+			}
+			// "Connection reset by peer" is usually a transient error.
```

Comment:
I'm seeing this error from both write and read operations in large tests. But sure - we can change it to retry only GET operations and fix tests to retry in tests in case of write operations.